cypress/integration/identity_creation.spec.js

```diff
@@ -12,6 +12,89 @@ context("identity creation", () => {
     });
   });
 
+  context("navigation", () => {
+    it("is possible to step through the identity creation flow", () => {
+      // Intro screen
+      cy.get('[data-cy="get-started-button"]').click();
+
+      // Enter details screen
+      cy.get('[data-cy="form"] [data-cy="handle"]').type("rafalca");
+      cy.get('[data-cy="form"] [data-cy="display-name"]').type(
+        "Rafalca Romney"
+      );
+      cy.get('[data-cy="create-id-button"]').click();
+
+      // Confirmation screen
+      // TODO(rudolfs): change the emoji once the backend returns the right one
+      cy.get('[data-cy="identity-card"] img[alt="🐽"]').should("exist");
+      cy.get('[data-cy="identity-card"]')
+        .contains("Rafalca Romney")
+        .should("exist");
+      cy.get('[data-cy="identity-card"]')
+        .contains("rafalca@123abcd.git")
+        .should("exist");
+
+      // Land on profile screen
+      cy.get('[data-cy="go-to-profile-button"]').click();
+      // TODO(rudolfs): change this to the actual handle that we
+      // just created once the backend is wired up
+      cy.get('[data-cy="profile-avatar"]').contains("cloudhead");
+    });
+
+    it("is possible to directly register your identity after creating it", () => {
+      cy.get('[data-cy="get-started-button"]').click();
+
+      cy.get('[data-cy="form"] [data-cy="handle"]').type("rafalca");
+      cy.get('[data-cy="create-id-button"]').click();
+      cy.get('[data-cy="register-identity-link"]').click();
+
+      cy.contains("Register your handle").should("exist");
+
+      cy.get('[data-cy="cancel-button"]').click();
+      // TODO(rudolfs): change this to the actual handle that we
+      // just created once the backend is wired up
+      cy.get('[data-cy="profile-avatar"]').contains("cloudhead");
+    });
+
+    context(
+      "when clicking cancel or modal close button before the identity is created",
+      () => {
+        it("sends the user back to the intro screen", () => {
+          cy.get('[data-cy="get-started-button"]').click();
+          cy.get('[data-cy="cancel-button"]').click();
+
+          // We should land back on the intro screen
+          cy.get('[data-cy="get-started-button"]').click();
+
+          // Now try to close the modal via the "x" button
+          cy.get('[data-cy="modal-close-button"]').click();
```

Comment:
Should there be a check that the modal is still open?

Reply:
Good catch.